cfgov/v1/models/base.py

```diff
@@ -255,6 +267,37 @@ def get_context(self, request, *args, **kwargs):
         if self.schema_json:
             context['schema_json'] = self.schema_json
 
+        context['meta_description'] = ''
+        if self.search_description:
+            context['meta_description'] = self.search_description
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'hero',
+                'body')
+        if (
+            hasattr(self, 'preview_description')
+            and not context['meta_description']
+        ):
+            context['meta_description'] = self.preview_description
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'text_introduction',
+                'intro')
+        if hasattr(self, 'content') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.content,
+                'text_introduction',
+                'intro')
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'item_introduction',
+                'paragraph')
```

Comment:
This may be in the reverse hierarchy intended. Each "if" section overwrites the previous one, if the sought content exists. Do we prefer search description if it exists, and then try to find other bits of content as fallbacks?

Reply:
It would be a little cleaner to make every `if` after the first one an `elif` and get rid of all the `and not context['meta_description']`s.

Reply:
Well, maybe not. Because that would require some gnarly constructions for the things we're looking for inside StreamField blocks.